utils/paste.tsx

```diff
@@ -66,8 +66,14 @@ export function formatMarkdownTableMessage(table: HTMLTableElement, message?: st
     }).join('\n');
 
     const formattedTable = `${header}${body}\n`;
-
-    return message ? `${message}\n\n${formattedTable}` : formattedTable;
+    if (!message) {
+        return formattedTable;
+    }
+    if (typeof caretPosition === 'undefined') {
+        return message ? `${message}\n\n${formattedTable}` : formattedTable;
+    }
+    const newMessage = [message.slice(0, caretPosition), formattedTable, message.slice(caretPosition)];
+    return newMessage.join('\n');
```

Comment:
Should this be a double `\n` to be consistent with the case above when no `caretPosition` is provided?

Reply:
originally I had `\n\n` but there were a lot of spaces around and it looked weird because the table is going to render in a new line.

I thought about changing the other one as well, but it probably is there because the most likely case is that the user didn't add it, and just pasted the table in, so having that `\n` might make everything more readable.

I'm 0/5 on this, so if you think it should have the double new-line, I really don't mind pushing the change.

Reply:
0/5, just caught my attention that it was missing the `/n`. I think it's fine that it could be slightly different in a case where it's pasted between existing text.